django/contrib/postgres/search.py

```diff
@@ -219,6 +221,53 @@ def as_sql(self, compiler, connection, function=None, template=None):
         return sql, extra_params + params
 
 
+class SearchHeadline(Func):
+    function = 'ts_headline'
+    output_field = TextField()
+    options = None
+
+    def __init__(self, expression, query, **extra):
+        if not hasattr(query, 'resolve_expression'):
+            query = SearchQuery(query)
+        self.config = query.config
```

Comment:
I couldn't really see the point of having to repeat the config when it's already set on the `SearchQuery`. And if I'm not missing something here specifying two different configs could give pretty weird results:

```python
SearchHeadline('dialogue', SearchQuery('cadeaux', config='french'), config='chinese')
```

Reply:
So do we need to generate SQL with the same config in the `ts_headline` and an inline `tsquery`? (that's my main question)
```sql
ts_headline('french'::regconfig,  ..., tsquery(..., 'french'::regconfig), ...)
```
If not then I would prefer to leave config only in `tsquery` and don't duplicate its logic in `SearchHeadline`.

Reply:
Yes AFAIK the config needs to be explicitly specified on `ts_headline` even if it already is set on the `tsquery`. If not specified it will fall back on the config set as `default_text_search_config`.

Reply:
I think we should support both, the main `config` is used for parsing a document and `tsquery`'s `config` is used for parsing a query. I would handle the main `config` in `SearchHeadline` and leave `SearchQuery()` as a expression.

Reply:
Sure that makes sense. Maybe we should always allow to set `config` which will override config from an eventual `SearchQuery ` when set?